backend/internal/storage/storage.go

```diff
@@ -28,7 +30,7 @@ type FeedbackComment interface {
 type Works interface {
 	GetWorks(ctx context.Context, classroomID int, assignmentID int) ([]*models.StudentWorkWithContributors, error)
 	GetWork(ctx context.Context, classroomID int, assignmentID int, studentWorkID int) (*models.PaginatedStudentWorkWithContributors, error)
-	CreateStudentWork(ctx context.Context, work *models.StudentWork, GHUserID int64) error
+	CreateStudentWork(ctx context.Context, assignmentOutlineID int32, gitHubUserID int64, repoName string, workState string, dueDate time.Time) (int, error)
```

Comment:
not the biggest fan of returning just an ID, rather than returning the created/modified object

Reply:
In my opinion, it is redundant. You provided all of the information to the create function, and the data is not being manipulated/changed. We return the ID since it is the only generated data field (besides `CreatedAt` which is not used).

Reply:
I think it follows RESTful conventions to return the updated/created item.

- It allows the frontend to immediately update the UI with accurate server-generated data, rather than optimistic changes based on the frontend implementation
- It confirms that the fields were processed correctly
- If there are any server-generated fields, it is required